Features/Context/SubContext/CommonActions.php

```diff
@@ -0,0 +1,363 @@
+
+        return $visibility;
+    }
+
+    /**
+     * Fills a sub form with values
+     *
+     * @param   string    $text
+     * @param   array     $values
+     */
+    protected function fillSubFormList( $text, $values )
+    {
+        $mainSelector = ".ez-editfield-infos";
+        $ancestorSelector = ".ez-authors-input-container";
+        $childSelector = ".ez-validated-input, .ez-field-sublabel";
+
+        $jsArgs = JsHelper::generateFuncArgs( $text, $mainSelector, $values, $ancestorSelector, $childSelector );
+        $jsCode = "return BDD.fillListPair( $jsArgs )";
+        $this->execJavascript( $jsCode );
+    }
+
+    /**
+     * Finds an HTML element by class and the text value and clicks it
+     *
+     * @param string $name
+     */
+    protected function clickElementsByText( $text, $selector )
+    {
+        $jsArgs = JsHelper::generateFuncArgs( $text, $selector );
+        $jsCode = "return BDD.clickElementByText( $jsArgs );";
+        $this->execJavascript( $jsCode );
+    }
+
+    /**
+     * Finds an HTML element by class and the text value and returns it
+     *
+     * @param string $name
+     */
+    protected function checksElementsByText( $text, $selector )
+    {
+        $jsArgs = JsHelper::generateFuncArgs( $text, $selector );
+        $jsCode = "return BDD.getElementByText( $jsArgs );";
+        return $this->evalJavascript( $jsCode );
+    }
+
+    /**
+     * Finds an HTML element by class and the text value and return the value of the sibling
+     *
+     * @param string $name
+     */
+    protected function checksElementSiblingValue( $text, $selector )
+    {
+        $jsArgs = JsHelper::generateFuncArgs( $text, $selector );
+        $jsCode = "return BDD.checksElementSiblingValueByText( $jsArgs );";
+        return $this->evalJavascript( $jsCode );
+    }
+
+    /**
+     * Finds an HTML element by class and the text value and clicks/hovers it
+     *
+     * @param string $name
+     */
+    protected function clickContentTreeLink( $dir, $index = 0, $rootSelectorId = '' )
+    {
+        $path = explode( "/", $dir );
+
+        $jsArgs = JsHelper::generateFuncArgs( $path, $index, $rootSelectorId );
+        $jsCode = "return BDD.openTreePath( $jsArgs );";
+
+        while ( ( $result = $this->evalJavascript( $jsCode ) ) === -1 )
+        {
+            usleep( 100 * 1000 ); // 100ms
+        }
+
+        if ( is_string( $result ) )
+        {
+            return $this->clickContentTreeLink( $dir, ++$index, $result );
+        }
+    }
+
+    /**
+     * Makes and element visible so a file can be attached to it
+     *
+     * @param string $path      file path relative to mink definitions
+     * @param string $selector  selector of the element do to make visible
+     */
+    protected function attachFilePrepare( $path, $selector )
+    {
+        $jsArgs = JsHelper::generateFuncArgs( $selector );
+        $jsCode = "return BDD.changeCssDisplay( $jsArgs );";
```

Comment:
hum that's a bit weird to have to unhide the file input to fill it, is that really needed ?